projects/core/src/occ/adapters/cms/occ-cms-component.adapter.ts

```diff
@@ -53,31 +69,78 @@ export class OccCmsComponentAdapter implements CmsComponentAdapter {
     };
 
     requestParams['componentIds'] = ids.toString();
-
-    return this.http
-      .get<Occ.ComponentList>(
-        this.getComponentsEndpoint(requestParams, fields),
-        {
-          headers: this.headers,
-        }
-      )
-      .pipe(
+    if (this.featureConfigService.isEnabled('newCmsEndpoint')) {
+      return this.userIdService.getUserId().pipe(
+        switchMap((userId: string) => {
+          return this.http.get<Occ.ComponentList>(
+            this.getComponentsEndpoint(requestParams, fields, userId),
+            {
+              headers: this.headers,
+            }
+          );
+        }),
         map((componentList) => componentList.component ?? []),
         this.converter.pipeableMany(CMS_COMPONENT_NORMALIZER)
       );
+    } else {
+      return this.http
+        .get<Occ.ComponentList>(
+          this.getComponentsEndpoint(requestParams, fields),
+          {
+            headers: this.headers,
+          }
+        )
+        .pipe(
+          map((componentList) => componentList.component ?? []),
+          this.converter.pipeableMany(CMS_COMPONENT_NORMALIZER)
+        );
+    }
   }
 
-  protected getComponentEndPoint(id: string, pageContext: PageContext): string {
-    return this.occEndpoints.buildUrl('component', {
-      urlParams: { id },
-      queryParams: this.getContextParams(pageContext),
-    });
+  protected getComponentEndPoint(
+    id: string,
+    pageContext: PageContext,
+    userId?: string
+  ): string {
+    if (this.featureConfigService.isEnabled('newCmsEndpoint')) {
+      const queryParams = this.getContextParams(pageContext);
+      const attributes = userId
+        ? {
+            urlParams: { id, userId },
+            queryParams,
+          }
+        : { urlParams: { id }, queryParams };
+
+      return this.occEndpoints.buildUrl('component', attributes);
+    } else {
+      return this.occEndpoints.buildUrl('component', {
+        urlParams: { id },
+        queryParams: this.getContextParams(pageContext),
+      });
+    }
   }
 
-  protected getComponentsEndpoint(requestParams: any, fields: string): string {
-    return this.occEndpoints.buildUrl('components', {
-      queryParams: { fields, ...requestParams },
-    });
+  protected getComponentsEndpoint(
+    requestParams: any,
+    fields: string,
+    userId?: string
+  ): string {
+    if (this.featureConfigService.isEnabled('newCmsEndpoint')) {
+      const queryParams = { fields, ...requestParams };
+
+      const attributes = userId
+        ? {
+            urlParams: { userId },
+            queryParams,
+          }
+        : { queryParams };
+
+      return this.occEndpoints.buildUrl('components', attributes);
+    } else {
+      return this.occEndpoints.buildUrl('components', {
```

Comment:
Also we can move this part out of else statement.

Reply:
fixed